src/class-permissions-helper.php

```diff
@@ -203,4 +224,27 @@ public function post_type_has_admin_bar( $post_type ) {
 	public function is_copy_allowed_to_be_republished( \WP_Post $post ) {
 		return \in_array( $post->post_status, [ 'dp-rewrite-republish', 'private' ], true );
 	}
+
+	/**
+	 * Determines if the post has a trashed copy intended for Rewrite & Republish.
+	 *
+	 * @param \WP_Post $post The post object.
+	 *
+	 * @return bool Whether the post has a trashed copy intended for Rewrite & Republish.
+	 */
+	public function has_trashed_rewrite_and_republish_copy( \WP_Post $post ) {
+		$copy_id = \get_post_meta( $post->ID, '_dp_has_rewrite_republish_copy', true );
+
+		if ( ! $copy_id ) {
+			return false;
+		}
+
+		$copy = \get_post( $copy_id );
```

Comment:
Perhaps for later: This would be a good candidate to be moved to the Utils class, where instead of returning a boolean, we return the copied post if it exists, otherwise `null`.

Then, here you could invert the `if` whether the Utils' result is empty or the post status of the copy is not `trash`.

Reply:
Yeah. Also, this method is very similar to the `has_scheduled_rewrite_and_republish_copy()` method which returns `bool|\WP_Post`. I'd agree some polishing to avoid almost-duplicate code would be good to do, maybe later.